### PR TITLE
Initial implementation of concurrency testing

### PR DESCRIPTION
**Pull request by jszwedko**
This updates the http_test_server to allow for additional configuration
to artifically control the behavior incoming requests. It allows for
testing vector (and other tools') behavior in the presence of rate
limits or other artifical contraints.

In addition to changes to the test server, this also introduces some
helper scripts to test vector's concurrency behavior under a variety of
scenarios. I expect we might expand these to allow running other tools
(like fluentbit) to observe concurrency behavior; possibly plotting them
all together.

It's probably easiest to review the README changes first. I attempted to maintain compatibility with current behavior to not break the test harness's expectations, but did make some pretty sweeping changes. A design goal here was to split up the test artifact processing into separate steps so that we can explore the result data adhoc and codify any other interesting views.

Signed-off-by: Jesse Szwedko <jesse@szwedko.me>

### CODE REVIEW

**Comment by jszwedko** [comment]
Example graph output (using apache bench with concurrency of 100):

concurrent_requests

File: go.mod [hunk 1]
```diff
@@ -2,4 +2,17 @@ module github.com/timberio/http_test_server
 
 go 1.13
 
-require github.com/urfave/cli v1.22.1
```
Comment (jszwedko) [inline go.mod:5]: I switched to `viper`/`cobra` given `urfave/cli` is not as well maintained these days and I wanted easier out-of-the-box configuration (like handling environment variables for flags).

**Review by binarylogic — APPROVED** [review]
Nice work! From everything I can tell this should not change the existing behavior, which is important for the test harness. Otherwise, I trust your Go experience much more than mine.

File: README.md [hunk 2]
```diff
@@ -1,4 +1,4 @@
-# HTTP Test Server
+Automated # HTTP Test Server
```
Comment (MOZGIII) [inline README.md]:
```suggestion
# Automated HTTP Test Server
```

**Comment by jszwedko** [comment]
@JeanMertz if you have some time, I wouldn't mind another pair of gopher eyes on this. I realize there are a lot of changes though. If you can review, I'd recommend going in the order:

* README
* `main.go`
* `server.go`
* the other go files
* `bin/*`

You could also review the commits in order if you liked. The first one is big, but get smaller after that.

**Pull request merged**